packages/embarkjs/src/storage.js

```diff
@@ -66,7 +66,7 @@ Storage.setProvider = function (providerName, options) {
 
 Storage.isAvailable = function () {
   if (!this.currentStorage) {
-    return false;
+    return Promise.resolve(false);
```

Comment:
@emizzle can you add a comment on why we're enforcing asynchronisity here? A comment here is enough, doesn't need to be in the source.

Reply:
Good point, I should have clarified.

When storage is disabled (in the config), `this.curentStorage` is `undefined`, and returning `false` to a caller who is expecting a `Promise` throws an error. For example, in the demo, the browser calls `EmbarkJS.Storage.isAvailable().then((isAvailable) => { ... })` which will throw an error on `.then()` when `.isAvailable()` doesn't return a Promise.

Does that make sense?